docs/eBPFWeakReferences.md

```diff
@@ -0,0 +1,134 @@
+
+**Introduction and problem statement**
+
+The eBPF user-kernel interface is built around a few key objects such as 'Programs', 'Maps' and their interaction. The
+'eBPF for Windows' implementation makes extensive use of 'reference counting' to manage the interactions and lifetimes
+of these objects.  Code paths 'acquire' references to the required objects for the needed duration and 'release' them
+when done.  The reference counting design ensures that the referred-to objects remain valid (i.e. are not deleted) for
```

Comment:
```suggestion
when done.  The reference counting design ensures that the referred-to objects remain valid (i.e., are not deleted) for
```
non-blocking nit: per Chicago Manual of Style, "i.e." and "e.g." should always have a comma after them, like other places in this file do.

Reply:
fixed